tests/test_intl/test_intl.py

```diff
@@ -681,50 +682,188 @@ def test_translation_progress_classes_true(app):
+    bom_rst = app.srcdir / 'bom.rst'
+    bom_rst_time = time.time_ns()
+
+    index_rst = app.srcdir / 'index.rst'
+    index_rst_time = time.time_ns()
+    po_time = time.time_ns()
+
+    # patch the 'creation time' of the source files
+    assert _set_mtime_ns(po_path, po_time) == po_time
+    assert _set_mtime_ns(bom_rst, bom_rst_time) == bom_rst_time
+    assert _set_mtime_ns(index_rst, index_rst_time) == index_rst_time
+
+    assert not mo_path.exists()
+    # when writing mo files, the counter is updated by calling
+    # patch_write_mo which is called to create .mo files (and
+    # thus the timestamp of the files are not those given by
+    # the OS but our fake ones)
+    app.build()
+    assert mo_path.exists()
+    # Do a real sleep on POSIX, or simulate a sleep on Windows
+    # to ensure that calls to time.time_ns() remain consistent.
+    clock.sleep(0.1 if os.name == 'posix' else 1)
+
+    # check that the source files were not modified
+    assert bom_rst.stat().st_mtime_ns == bom_rst_time
+    assert index_rst.stat().st_mtime_ns == index_rst_time
+    # check that the 'bom' document is discovered after the .mo
+    # file has been written on the disk (i.e., read_doc() is called
+    # after the creation of the .mo files)
+    assert app.env.all_docs['bom'] > mo_path.stat().st_mtime
 
-    # phase1: build document with non-gettext builder and generate mo file in srcdir
-    app0 = make_app('dummy', *args, **kwargs)
-    app0.build()
-    time.sleep(0.01)
-    assert (app0.srcdir / 'xx' / 'LC_MESSAGES' / 'bom.mo').exists()
     # Since it is after the build, the number of documents to be updated is 0
-    update_targets = get_update_targets(app0)
-    assert update_targets[1] == set(), update_targets
+    update_targets = _get_update_targets(app)
+    assert update_targets[1] == set()
     # When rewriting the timestamp of mo file, the number of documents to be
     # updated will be changed.
-    mtime = (app0.srcdir / 'xx' / 'LC_MESSAGES' / 'bom.mo').stat().st_mtime
-    os.utime(app0.srcdir / 'xx' / 'LC_MESSAGES' / 'bom.mo', (mtime + 5, mtime + 5))
-    update_targets = get_update_targets(app0)
-    assert update_targets[1] == {'bom'}, update_targets
+    new_mo_time = time.time_ns()
+    assert _set_mtime_ns(mo_path, new_mo_time) == new_mo_time
+    update_targets = _get_update_targets(app)
+    assert update_targets[1] == {'bom'}
+    mock.undo()  # explicit call since it's not a context
 
-    # Because doctree for gettext builder can not be shared with other builders,
-    # erase doctreedir before gettext build.
-    shutil.rmtree(app0.doctreedir)
+    # remove all sources for the next test
+    shutil.rmtree(app.srcdir, ignore_errors=True)
+    time.sleep(0.1 if os.name == 'posix' else 0.5)  # real sleep
+
+
+@sphinx_intl
+@pytest.mark.sphinx('gettext', testroot='builder-gettext-dont-rebuild-mo', freshenv=True)
+def test_gettext_dont_rebuild_mo(mock_time_and_i18n, app):
```

Comment:
In some ways it makes sense to re-separate these tests; however I think the original intent was to confirm that after using one builder to create a `.mo` file, the second builder _does not_ rebuild that exact same `.mo` file.

Reply:
We are actually deleting the doctrees by the first one. We use the dummy builder to actually write the po files but that could be done separately. 

Previously, we checked that the dummy builder did indeed rebuild the files and that the gettext does not. But since we only care about the file but not its content nor from where it comes, it shouldn't matter.